src/core_read.cpp

```diff
@@ -74,16 +74,27 @@ CScript ParseScript(const std::string &s) {
         if (std::all_of(w.begin(), w.end(), ::IsDigit) ||
             (w.front() == '-' && w.size() > 1 &&
              std::all_of(w.begin() + 1, w.end(), ::IsDigit))) {
+            bool is_valid = true;
+            int64_t n;
             // Number
-            int64_t n = atoi64(w);
+            try {
+                n = std::stoll(w);
+            } catch (...) {
+                is_valid = false;
+            }
+
+            if (n == std::numeric_limits<int64_t>::min()) {
```

Comment:
If w doesn't parse, why do we want to do this check?

Reply:
Cuz -0x8000000000000000 is a valid int64_t but an invalid script number

Reply:
Yes. It just seems like if the parsing fails there's no reason to *also* do this check. Couldn't it be moved into the try block?

Reply:
Can you propose a suggestion? My C++ is not good enough for that

Reply:
I think I got it. This should be what you meant.